Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Beckn/ACL/Select.hs

```diff
@@ -107,27 +72,31 @@ buildSelectReqV2 subscriber req = do
         bapId = subscriber.subscriber_id,
         bapUri = subscriber.subscriber_url,
         pickupTime = now,
-        autoAssignEnabled = fromMaybe False autoAssignEnabled,
+        autoAssignEnabled = autoAssignEnabled,
         customerExtraFee = customerExtraFee,
         estimateId = Id estimateIdText
       }
 
-getCustomerExtraFee :: Select.TagGroups -> Maybe Money
-getCustomerExtraFee tagGroups = do
-  tagValue <- getTag "customer_tip_info" "customer_tip" tagGroups
-  customerExtraFee <- readMaybe $ T.unpack tagValue
-  Just $ Money customerExtraFee
-
 getCustomerExtraFeeV2 :: [Spec.TagGroup] -> Maybe Money
 getCustomerExtraFeeV2 tagGroups = do
   tagValue <- Utils.getTagV2 Tag.CUSTOMER_TIP_INFO Tag.CUSTOMER_TIP tagGroups
   customerExtraFee <- readMaybe $ T.unpack tagValue
   Just $ Money customerExtraFee
 
-getAutoAssignEnabledV2 :: [Spec.TagGroup] -> Maybe Bool
-getAutoAssignEnabledV2 tagGroups = do
-  tagValue <- Utils.getTagV2 Tag.AUTO_ASSIGN_ENABLED Tag.IS_AUTO_ASSIGN_ENABLED tagGroups
-  case tagValue of
-    "True" -> Just True
-    "False" -> Just False
-    _ -> Just False
+getAutoAssignEnabledV2 :: Maybe [Spec.TagGroup] -> Bool
+getAutoAssignEnabledV2 = \case
+  Nothing -> True -- By default we only need to send one quote
+  Just tagGroups ->
+    let tagValue = Utils.getTagV2 Tag.AUTO_ASSIGN_ENABLED Tag.IS_AUTO_ASSIGN_ENABLED tagGroups
+     in case tagValue of
+          Just "True" -> True
+          Just "False" -> False
+          _ -> False
+
+cacheSelectMessageId :: CacheFlow m r => Text -> Text -> m ()
+cacheSelectMessageId messageId transactionId = do
+  let msgKey = mkTxnIdKey transactionId
+  Hedis.setExp msgKey messageId 3600
+
+mkTxnIdKey :: Text -> Text
```

Comment:
This can be made common? 
Having this function in two places can cause mismatch issues

Reply:
fixed.